### PR TITLE
fix issue in SQS instrumentation when no MessageAttributes are set

### LINKED ISSUE #1572: Error in instrumentation on boto3 sqs client function send_message_batch
**Bug**
Error in instrumentation on boto3 sqs client function send_message_batch when do not define non required parameter MessageAttributes in entries.

[SQS.Client.send_message_batch](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sqs.html#SQS.Client.send_message_batch)

**To Reproduce**
instrumet code
start a transaction (without a started transaction code works fine)
send_bactch_message without Entries.MessageAttributes parameter

```
import uuid
import boto3
import elasticapm
from elasticapm import Client


if __name__ == '__main__':
    elasticapm.instrument()
    apm_client = Client(service_name=apm_info['SERVICE_NAME'],
                        server_url=apm_info['SERVER_URL'],
                        secret_token=apm_info['SECRET_TOKEN'],
                        environment=apm_info['Environment_name'])

    elasticapm.instrument()
    apm_client.begin_transaction(transaction_type="script")
    try:
        sqs_client = boto3.client('sqs')
        sqs_url = MY_SQS_URL
        entries = [dict(Id=uuid.uuid4().hex, MessageBody=str(i)) for i in range(10)]
        sqs_client.send_message_batch(QueueUrl=sqs_url, Entries=entries)
    except:
        apm_client.apm_client.capture_exception()
    apm_client.end_transaction(name="producer", result='success')
```
**workaround**
Definining Parameter Entries.MessageAttributes={} instrumentation works fine.

```
import uuid
import boto3
import elasticapm
from elasticapm import Client


if __name__ == '__main__':
    elasticapm.instrument()
    apm_client = Client(service_name=apm_info['SERVICE_NAME'],
                        server_url=apm_info['SERVER_URL'],
                        secret_token=apm_info['SECRET_TOKEN'],
                        environment=apm_info['Environment_name'])

    elasticapm.instrument()
    apm_client.begin_transaction(transaction_type="script")
    try:
        sqs_client = boto3.client('sqs')
        sqs_url = MY_SQS_URL
        entries = [dict(Id=uuid.uuid4().hex, MessageBody=str(i), MessageAttributes={}) for i in range(10)]
        sqs_client.send_message_batch(QueueUrl=sqs_url, Entries=entries)
    except:
        apm_client.apm_client.capture_exception()
    apm_client.end_transaction(name="producer", result='success')
```

`
**Environment (please complete the following information)**
- OS: Release Linux Mint 20.3 Una 64-bit, Kernel Linux 5.13.0-41-generic x86_64, MATE 1.26.0
- Python version: 3.8.10 
- APM Server version: 8.1.2
- Agent version: elastic-apm==6.9.1

**Additional context**

- Agent config options 
  <details>
    <summary>Click to expand</summary>

```
Traceback (most recent call last):
  File "/home/user/test/python/eapm_bug//tests/apm_bug.py", line 25, in <module>
    sqs_client.send_message_batch(QueueUrl=sqs_url, Entries=entries)
  File "/home/user/test/python/eapm_bug//venv/lib/python3.8/site-packages/botocore/client.py", line 386, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/home/user/test/python/eapm_bug//venv/lib/python3.8/site-packages/elasticapm/instrumentation/packages/base.py", line 210, in call_if_sampling
    return self.call(module, method, wrapped, instance, args, kwargs)
  File "/home/user/test/python/eapm_bug//venv/lib/python3.8/site-packages/elasticapm/instrumentation/packages/botocore.py", line 93, in call
    span_modifiers[service](span, args, kwargs)
  File "/home/user/test/python/eapm_bug//venv/lib/python3.8/site-packages/elasticapm/instrumentation/packages/botocore.py", line 194, in modify_span_sqs
    if len(message["MessageAttributes"]) + attributes_count <= SQS_MAX_ATTRIBUTES:
KeyError: 'MessageAttributes'

Process finished with exit code 1

```
  </details>
- `requirements.txt`:
  <details>
    <summary>Click to expand</summary>

    ```
    elastic-apm[flask]>=6.8.1,<7
    ```
  </details>

### PR DESCRIPTION
**Pull request by beniwohli**
## Related issues
closes #1572

### CODE REVIEW

File: elasticapm/instrumentation/packages/botocore.py [hunk 1]
```diff
@@ -214,6 +214,7 @@ def modify_span_sqs(span, args, kwargs):
         else:
             messages = []
         for message in messages:
+            message["MessageAttributes"] = message.get("MessageAttributes") or {}
```
Comment (basepi) [inline elasticapm/instrumentation/packages/botocore.py:217]: Is `message` not a normal dict? Or is the key defined as `None`, maybe? Just thought it looked a little odd, but I certainly might be missing something.

```suggestion
            message["MessageAttributes"] = message.get("MessageAttributes", {})
```

Comment (beniwohli) [reply elasticapm/instrumentation/packages/botocore.py:217]: It's just a silly little optimization. The thinking was that with `.get("foo", {})`, a new `dict` is always instantiated and subsequently thrown away if the key is found. With `or`, the dict is only instantiated if the lookup fails, thanks to short-circuiting. A quick benchmark using `timeit`:

    In [1]: message = {"a": 1}
    In [2]: never_get = lambda: message.get("a", {})
    In [3]: never_or = lambda: message.get("a") or {}
    In [4]: always_get = lambda: message.get("b", {})
    In [5]: always_or = lambda: message.get("b") or {}
    In [6]: import timeit
    In [7]: timeit.timeit(never_get, setup="gc.enable()")
    Out[7]: 0.09470749995671213
    In [8]: timeit.timeit(never_or, setup="gc.enable()")
    Out[8]: 0.08076889999210835
    In [9]: timeit.timeit(always_get, setup="gc.enable()")
    Out[9]: 0.09426169993821532
    In [10]: timeit.timeit(always_or, setup="gc.enable()")
    Out[10]: 0.09646200004499406

So, using `or` is somewhat faster when the key is there most of the time. It's only a difference of ~0.01us though, so if you prefer the slightly less confusing code, I'm ok with that.

Comment (basepi) [reply elasticapm/instrumentation/packages/botocore.py:217]: Interesting! Thanks for the explanation!

**Comment by julianogv** [comment]
@beniwohli any ETA on releasing thix fix to pypi?

**Pull request merged**